README.md

```diff
@@ -1,2 +1,119 @@
+
+# Identity
+
+An identity system is an important consideration in social network design as it affects security, user experience, and social dynamics. Pseudonymous-name networks like Reddit encourage different user behaviors from real-name networks like Facebook. Our goal with Farcaster is to design an identity system that is: 
+
+1. **Secure, decentralized, and pseudonymous**:  Users must be able to claim and use an identity on the network without third-party approval.
+
+2. **User-friendly**: Acquiring an identity should be quick, cheap, and easy. In practical terms, users should be able to get started in under a minute for less than $10, without any special knowledge about blockchains or wallets.
+
+3. **Trustworthy**: Squatting and impersonation of scarce username resources should be addressed to ensure that they are owned by honest, active users. 
+
+On Farcaster, users establish identity by claiming a **Farcaster Identification Number (FIN)**. A FIN serves as a pointer to an account, similar to a primary key in a database. Users may have many social accounts and can get a FIN for each one. Users can optionally connect a FIN with an NFT username they own, like an ENS or [Farcaster name](#names). While identifiers can be acquired freely and in a decentralized manner, user namespaces are free to implement their own rules and regulations.
+
+Applications should always reference a user account by its FIN and never by username or display name. The FIN is the only identifier that is guaranteed to be decentralized and always point to the same account. Applications can replace the FIN with the currently associated display or verified username at render time. User accounts become easier to recognize and harder to impersonate when displayed this way, and users still retain the ability to change their username in the future. This design meets our goals and, as a side effect, addresses the [Zooko's Triangle](https://en.wikipedia.org/wiki/Zooko%27s_triangle) naming trilemma. 
+
+## Identification Number
+
+A Farcaster Identification Number (FIN) is a deterministically generated number like `324234`. FINs are issued by the Farcaster Identity contract on the Ethereum blockchain, which ensures that an address can only hold one FIN at a time. FINs can be transferred between addresses, as long as this invariant is held. FINs begin at 0 and increment by 1 with every new identity issued, which is the most gas efficient way to issue unique identifiers on the EVM. They have a max size equal to the maximum value of a `uint256`, which is practically infinite for our needs. 
```

Comment:
I find `our needs` a bit vague is there a more direct way to explain like "representing all possible future user identifications"